### PR TITLE
implements transfer progress bar

### PR DESCRIPTION
**Pull request by EdwinFairchild**
I implemented this because the current output overflows when using 2M.
see it in action  here https://media.giphy.com/media/VP4Ua0yKmRa0YYv5UH/giphy.gif

Could be that less is more sometimes, I can make it just simply print percent line after line. 
Your call.

### CODE REVIEW

File: Examples/MAX32655/BLE_otac/datc_main.c [hunk 1]
```diff
@@ -768,9 +771,12 @@ static void datcBtnCback(uint8_t btn)
                 /* Start the WDXC data stream */
                 datcCb.blockOffset[connId - 1] = BLOCK_OFFSET_INIT;
 
-                /* Put file request */
-                WdxcFtcSendPutReq(connId, datcCb.fileList[connId - 1][0].handle, BLOCK_OFFSET_INIT,
-                                  FILE_SIZE, BLOCK_OFFSET_INIT + FILE_SIZE, 0);
+                /* Put file request only if file has been discovered*/
+                if (fileDiscovered)
+                    WdxcFtcSendPutReq(connId, datcCb.fileList[connId - 1][0].handle,
```
Comment (lorne-maxim) [inline Examples/MAX32655/BLE_otac/datc_main.c:776]: clang-format should catch this, but you'll need braces around the body of this 'if' block.

File: Examples/MAX32655/BLE_otas/wdxs_file.c [hunk 2]
```diff
@@ -224,7 +235,21 @@ static uint8_t wdxsFileWrite(const uint8_t *pBuf, uint8_t *pAddress, uint32_t si
...
-        APP_TRACE_INFO2("Ext Flash: Wrote %d bytes @ 0x%08x", size, pAddress);
+        packetCount++;
+        /* print transfer progress */
+        if (packetCount % (modFileSize) == 0) {
+            progressbar[percentComplete] = '#';
+            APP_TRACE_INFO2("\033[1A%s%d%% Complete", progressbar, percentComplete * 10);
+            percentComplete++;
+        }
+        /* done receiving*/
+        if (packetCount >= (fileHeader.fileLen / 224) + 1) {
```
Comment (lorne-maxim) [inline Examples/MAX32655/BLE_otas/wdxs_file.c:246]: Add comment on why 224 or use #define.

**Review by lorne-maxim — CHANGES_REQUESTED** [review]

**Review by sihyung-maxim — COMMENTED** [review]
Did you merge the latest changes from main to this branch? Not sure why the linter is complaining, but maybe updating this branch will resolve the linter errors.

**Comment by EdwinFairchild** [comment]
@sihyung-maxim yeah when #192 get merged ill merge main into this again and it should clear things up

**Comment by Jake-Carter** [comment]
> @sihyung-maxim yeah when #192 get merged ill merge main into this again and it should clear things up

@EdwinFairchild merged

**Pull request closed without merging**